### PR TITLE
Add QAT codec for shuffle compression

### PR DESCRIPTION
**Pull request by marin-ma**

### CODE REVIEW

**Comment by zhanglistar** [comment]
What's the result comparing to lz4?

File: cpp/core/CMakeLists.txt [hunk 1]
```diff
@@ -264,6 +264,17 @@ if(ENABLE_HBM)
...
+  find_library(QATZIP_LIBRARY qatzip)
+  if(MEMKIND_LIBRARY STREQUAL "QATZIP_LIBRARY-NOTFOUND")
+    message(FATAL_ERROR "QATZip Library Not Found")
+  endif()
+  message(STATUS "Found QATZip Library: ${QATZIP_LIBRARY}")
+  target_sources(spark_columnar_jni PRIVATE utils/qat_util.cc)
+  target_link_libraries(spark_columnar_jni PUBLIC ${QATZIP_LIBRARY})
+  add_definitions(-DGLUTEN_ENABLE_QAT)
+endif()
+
```
Comment (FelixYBW) [inline cpp/core/CMakeLists.txt:277]: Does the script install the qat library automatically? If not, we may need to add the way to pre-install qat.

**Comment by FelixYBW** [comment]
Let's add a readme to briefly describe how to config qat as well. Link to qatzip library and Intel's QAT document for details.

File: cpp/core/CMakeLists.txt (hunk 1 above)
Comment (marin-ma) [reply cpp/core/CMakeLists.txt:277]: We need to both build and install QAT driver and QATzip. For QAT driver, we've only tested with the internal version. The public version can be downloaded [here](https://www.intel.com/content/www/us/en/download/19678/intel-quickassist-technology-intel-qat-driver-for-windows-for-intel-server-boards-and-systems-based-on-intel-62x-chipset.html), but the version is old.

**Comment by marin-ma** [comment]
> What's the result comparing to lz4?

We're still working on collecting the performance data. Once this is done, we will update in the doc.

**Review by FelixYBW — APPROVED** [review]

**Comment by FelixYBW** [comment]
OK to me. THank you, @marin-ma

**Pull request merged**